lib/Doctrine/ORM/Tools/SchemaTool.php

```diff
@@ -274,6 +275,15 @@ public function getSchemaFromMetadata(array $classes)
 
             if (isset($class->table['uniqueConstraints'])) {
                 foreach ($class->table['uniqueConstraints'] as $indexName => $indexData) {
+                    $uniqIndex = new Index($indexName, $indexData['columns'], true, false, [], isset($indexData['options']) ? $indexData['options'] : []);
+
+                    foreach ($table->getIndexes() as $tableIndexName => $tableIndex) {
+                        if ($tableIndex->isFullfilledBy($uniqIndex)) {
+                            $table->dropIndex($tableIndexName);
+                            break;
```

Comment:
@webimpress nothing denies adding more than one unique constraint for the same column: